lib/Ast.ml

```diff
@@ -876,15 +876,19 @@ end = struct
 
   let ( == ) = Base.phys_equal
 
-  let dump fs ctx ast =
+  let dump ctx ast fs =
     Format.fprintf fs "ast: %a@\nctx: %a@\n" T.dump ast T.dump ctx
 
-  let fail ctx ast exc =
-    if Conf.debug then (
-      let bt = Caml.Printexc.get_backtrace () in
-      dump Format.err_formatter ctx ast ;
-      Format.eprintf "%s%!" bt ) ;
-    raise exc
+  let assert_no_raise ~f ~dump x =
+    assert (
+      try
+        ignore (f x) ;
+        true
+      with exc ->
+        let bt = Caml.Printexc.get_backtrace () in
+        dump x Format.err_formatter ;
+        Format.eprintf "%s%!" bt ;
+        raise exc )
```

Comment:
`Conf.debug` was actually not needed, this code is only reachable when `assert` are enabled. I changed the code to make this clear.